go/nbs/persisting_chunk_source.go

```diff
@@ -8,24 +8,31 @@ import (
 	"bytes"
 	"io"
 	"sync"
+	"time"
 
 	"github.com/attic-labs/noms/go/chunks"
 	"github.com/attic-labs/noms/go/d"
 )
 
-func newPersistingChunkSource(mt *memTable, haver chunkReader, p tablePersister, rl chan struct{}) *persistingChunkSource {
+func newPersistingChunkSource(mt *memTable, haver chunkReader, p tablePersister, rl chan struct{}, stats *Stats) *persistingChunkSource {
+	t1 := time.Now()
+
 	ccs := &persistingChunkSource{mt: mt}
 	ccs.wg.Add(1)
 	rl <- struct{}{}
 	go func() {
 		defer ccs.wg.Done()
-		cs := p.Persist(mt, haver)
+		cs := p.Persist(mt, haver, stats)
 
 		ccs.mu.Lock()
 		defer ccs.mu.Unlock()
 		ccs.cs = cs
 		ccs.mt = nil
 		<-rl
+
+		if cs.count() > 0 {
+			stats.PersistLatency.SampleTime(time.Since(t1))
```

Comment:
the way this is right now, the time since `t1` will include any time spent waiting on the rate limiter. That may be desirable, or not. On one hand, it tells us how long we had to wait for this table to get done. On the other, we may be more interested in the time it takes to actually do the persisting (getting the chunks all compressed and written and yadda yadda)